src/generic/numberparsing.h

```diff
@@ -521,21 +1431,58 @@ really_inline bool parse_number(UNUSED const uint8_t *const src,
         // Ok, chances are good that we had an overflow!
         // this is almost never going to get called!!!
         // we start anew, going slowly!!!
-        return parse_float(src, parser, found_minus);
+        double d;
+        if(parse_float_strtod((const char*)src, &d)) {
+          parser.on_number_double(d);
+#ifdef JSON_TEST_NUMBERS // for unit testing
+          found_float(d, src);
+#endif
+          return true;
+        }
+#ifdef JSON_TEST_NUMBERS // for unit testing
+        found_invalid_number(src);
+#endif
+        return false;
       }
     }
-    if (unlikely((power_index > 2 * 308))) { // this is uncommon!!!
+    if (unlikely(exponent < FASTFLOAT_SMALLEST_POWER) ||
+      (exponent > FASTFLOAT_LARGEST_POWER)) {// this is uncommon!!!
       // this is almost never going to get called!!!
       // we start anew, going slowly!!!
-      return parse_float(src, parser, found_minus);
+        double d;
+        if(parse_float_strtod((const char *)src, &d)) {
+          parser.on_number_double(d);
+#ifdef JSON_TEST_NUMBERS // for unit testing
+          found_float(d, src);
+#endif
+          return true;
+        }
+#ifdef JSON_TEST_NUMBERS // for unit testing
+        found_invalid_number(src);
+#endif
+        return false;
     }
-    double factor = power_of_ten[power_index];
-    factor = negative ? -factor : factor;
-    double d = i * factor;
-    parser.on_number_double(d);
+    double d = 0;
+    bool success = true;
+    if (i != 0) {
+      d = compute_float_64(exponent, i, negative, &success);
+      if (!success) {
+        // we are almost never going to get here.
+        success = parse_float_strtod((const char *)src, &d);
+      }
+    }
+    if(success) {
+      parser.on_number_double(d);
+#ifdef JSON_TEST_NUMBERS // for unit testing
+      found_float(d, src);
+#endif
+      return true;
+    } else {
 #ifdef JSON_TEST_NUMBERS // for unit testing
-    found_float(d, src);
+      found_invalid_number(src);
```

Comment:
Not related here, just a comment: I've been thinking we could make a parser.on_invalid_number for this and then we wouldn't have to have an ifdef anymore--just hook up your own custom parser callback.

Reply:
I agree, but I don't want to handle this issue in this PR.